internal/models/session.go

```diff
@@ -0,0 +1,133 @@
+package models
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// SessionStorage is a struct to temporarily saving the session.
+type SessionStorage struct {
+	configuration SessionStorageConfiguration
+	sessions      map[string]*Session
+	mtx           sync.RWMutex
+	logger        logging.Logger
+}
+
+// NewSessionStorage creates a new temporary session and saves for validation.
+func NewSessionStorage(logger logging.Logger, config SessionStorageConfiguration) *SessionStorage {
+	ss := &SessionStorage{
+		configuration: config,
+		logger:        logger,
+		sessions:      make(map[string]*Session),
+	}
+
+	// session is deleted at MaxTTL (configurable).
+	go func() {
+		for now := range time.Tick(time.Second * 5) {
+			ss.mtx.Lock()
+			for key, value := range ss.sessions {
+				if now.Unix()-value.lastAccess > int64(config.MaxTTL) {
+					delete(ss.sessions, key)
+				}
+			}
+			ss.mtx.Unlock()
+		}
+	}()
+
+	return ss
+}
+
+// CreateSession creates a new session and saves it to memory.
+func (ss *SessionStorage) CreateSession() *Session {
```

Comment:
the point of a storage is to handle access to it, so store and retrieve, not to create